collections/soldiers.js

```diff
@@ -26,4 +26,36 @@ function newSoldier(data, callback) {
   }
 }
 
-module.exports.newSoldier = newSoldier;
\ No newline at end of file
+function getSoldier(name, id, callback) {
+  MongoClient.connect(url, function (err, client) {
+    assert.equal(null, err);
```

Comment:
the assert is elegant but not what we wish. this would crush the server. the node way to do this is to simply callback(err) this would have a similar impact but in a more programer controlled way